### PR TITLE
chat flow with assistant

### PR DESCRIPTION
**Pull request by chw-microsoft**
# Description

Add sample flow with inital version of assistant tool into test samples

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [x] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+   - `get_calorie_by_jogging(duration, temperature)``: calculate calorie for jogging exercise
+   - `get_calorie_by_jogging(duration, temperature)``: calculate calorie for swimming exercise
+
+## Prerequisites
+
+Install promptflow sdk and other dependencies in this folder:
+```sh
+pip install -r requirements.txt
+```
+Note: You need install internal pypi: https://msdata.visualstudio.com/Vienna/_build/results?buildId=115005260&view=artifacts&pathAsName=false&type=publishedArtifacts
```
Comment (weedqian) [inline src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/README.md:18]: Remove the internal pypi link?

File: src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+    print(f"Assistant_id: {assistant_id}, thread_id: {thread_id}, run_id: {run.id}")
+    return run
+
+
+@trace
+async def wait_for_status_check():
+    await asyncio.sleep(RUN_STATUS_POLLING_INTERVAL_IN_MILSEC / 1000.0)
+
+
+@trace
```
Comment (weedqian) [inline src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py:104]: It's not a good idea to add trace on wait_for_status_check and get_run_status. 

They are too noisy and meaningless.

File: src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+async def require_responses(cli: AsyncOpenAI, thread_id: str, run, invoker: AssistantToolInvoker):
+    tool_outputs = await get_tool_calls_outputs(invoker, run)
+    await submit_tool_calls_outputs(cli, thread_id, run.id, tool_outputs)
+
+
+@trace
+async def manage_run(cli: AsyncOpenAI, thread_id: str, invoker: AssistantToolInvoker, run):
```
Comment (weedqian) [inline src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py:144]: manage_run -> wait_for_run_complete?

File: src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        assistant = await create_assistant(cli, assistant_definition)
+        assistant_id = assistant.id
+
+    await add_message(cli, message, thread_id)
+
+    run = await start_run(cli, assistant_id, thread_id, assistant_definition, invoker)
+
+    await wait_for_run_complete(cli, thread_id, invoker, run)
+
+    # await get_run_steps(cli, thread_id, run.id)
```
Comment (thy09) [inline src/promptflow/tests/test_configs/flows/chat-with-calorie-assistant/add_message_and_run.py:41]: Remove the comment.

**Review by thy09 — APPROVED** [review]

**Review by thy09 — APPROVED** [review]

**Review by weedqian — APPROVED** [review]

**Pull request merged**